ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4301,6 +4301,19 @@ public void recycleDirToCmPath(Path dataPath, boolean isPurge) throws HiveExcept
     }
   }
 
+  private static void deleteAndRename(FileSystem destFs, Path destFile, FileStatus srcStatus, Path destPath)
+          throws IOException {
+    if (destFs.exists(destFile)) {
+      // rename cannot overwrite non empty destination directory, so deleting the destination before renaming.
+      destFs.delete(destFile);
```

Comment:
In non DistributedFileSystem case, we were deleting with recursive flag as false. But here we always keep it true. Does it have any impacts?

Reply:
anyways we want to delete the destination ..so i don't think any impact of recursive on this

Reply:
ok